### PR TITLE
[5610][next] Update the Pull from Remote dialog

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/5610

### CODE REVIEW

**Comment by jvega190** [comment]
@rart need to discuss the handling of errors of fetch site API

File: ui/app/src/services/sites.ts [hunk 1]
```diff
@@ -126,3 +126,7 @@ export function validateActionPolicy(
     }
   ).pipe(pluck(...toPluck));
 }
+
+export function fetchSite(siteId: string): Observable<any> {
+  return get(`/studio/api/1/services/api/1/site/get.json?site_id=${siteId}`).pipe(pluck('response'));
+}
```
Comment (rart) [inline ui/app/src/services/sites.ts]:
```suggestion
export function fetchLegacySite(siteId: string): Observable<LegacySite> {
  return get(`/studio/api/1/services/api/1/site/get.json?site_id=${siteId}`).pipe(pluck('response'));
}
```

**Review by rart — CHANGES_REQUESTED** [review]
Let's move all of the sandbox branch retrieval stuff to the PullDialogContainer

File: ui/app/src/components/GitManagement/PullDialog/PullDialogContainer.tsx [hunk 2]
```diff
@@ -30,47 +30,65 @@ import PrimaryButton from '../../PrimaryButton';
...
 } from '../../../utils/state';
 import useActiveSite from '../../../hooks/useActiveSite';
 import useActiveUser from '../../../hooks/useActiveUser';
+import Alert from '@mui/material/Alert';
+import FormLabel from '@mui/material/FormLabel';
+import { Typography } from '@mui/material';
+import Link from '@mui/material/Link';
 
-const useStyles = makeStyles(() =>
+const useStyles = makeStyles((theme) =>
```
Comment (rart) [inline ui/app/src/components/GitManagement/PullDialog/PullDialogContainer.tsx]: Update your code and make sure to use `makeStyles` from `tss-react` and remove `createStyles`.

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/services/sites.ts (hunk 1 above)
Comment (rart) [reply ui/app/src/services/sites.ts]: Changing the function name was not done

File: ui/app/src/services/sites.ts [hunk 3]
```diff
@@ -126,3 +126,7 @@ export function validateActionPolicy(
     }
   ).pipe(pluck(...toPluck));
 }
+
+export function fetchSite(siteId: string): Observable<LegacySite> {
```
Comment (rart) [inline ui/app/src/services/sites.ts]: Change name as requested in last review

File: ui/app/src/components/GitManagement/PullDialog/PullDialogContainer.tsx [hunk 4]
```diff
@@ -178,12 +199,22 @@ export function PullDialogContainer(props: PullFromRemoteDialogContainerProps) {
             ))}
           </Select>
         </FormControl>
+
+        <Alert severity="info" variant="outlined" className={classes.pullInfo}>
+          <Typography>
+            <FormattedMessage
+              id="repositories.pullInfo"
```
Comment (rart) [inline ui/app/src/components/GitManagement/PullDialog/PullDialogContainer.tsx]:
```suggestion
              id="repositories.pullFromCreatedBranchOnlyHint"
```

File: ui/app/src/components/GitManagement/PullDialog/PullDialogContainer.tsx [hunk 5]
```diff
@@ -137,27 +147,38 @@ export function PullDialogContainer(props: PullFromRemoteDialogContainerProps) {
...
+              <>
+                <Typography color="textSecondary" variant="body1" className={classes.pullBranchLabelHeading}>
+                  <FormattedMessage id="words.branch" defaultMessage="Branch" />
+                </Typography>
+                {sandboxBranch}
+              </>
+            ) : (
+              <Typography variant="body1" className={classes.pullBranchLabelError}>
+                <FormattedMessage
+                  id="repositories.sandboxBranchError"
```
Comment (rart) [inline ui/app/src/components/GitManagement/PullDialog/PullDialogContainer.tsx]:
```suggestion
                  id="repositories.sandboxBranchRetrievalError"
```

File: ui/app/src/components/GitManagement/PullDialog/PullDialogContainer.tsx [hunk 6]
```diff
@@ -58,16 +78,15 @@ export function PullDialogContainer(props: PullFromRemoteDialogContainerProps) {
     isSubmitting = false
   } = props;
 
-  const [selectedBranch, setSelectedBranch] = useState('');
   const [selectedMergeStrategy, setSelectedMergeStrategy] = useState('');
   const { classes } = useStyles();
   const { uuid, id: siteId } = useActiveSite();
   const { username } = useActiveUser();
+  const [sandboxBranch, setSandboxBranch] = useState('');
+  const [sandboxBranchError, setSandboxBranchError] = useState<ApiResponse>(null);
```
Comment (rart) [inline ui/app/src/components/GitManagement/PullDialog/PullDialogContainer.tsx]: merge these into a _spread state_ and add a loading flag and implement it in the brach display area.

File: ui/app/src/components/GitManagement/PullDialog/PullDialogContainer.tsx [hunk 7]
```diff
@@ -58,16 +78,15 @@ export function PullDialogContainer(props: PullFromRemoteDialogContainerProps) {
...
 
   const onFormInputChange = (e: any) => {
-    if (e.target.name === 'branch') {
-      setSelectedBranch(e.target.value);
-    } else if (e.target.name === 'mergeStrategy') {
+    if (e.target.name === 'mergeStrategy') {
       setSelectedMergeStrategy(e.target.value);
     }
```
Comment (rart) [inline ui/app/src/components/GitManagement/PullDialog/PullDialogContainer.tsx]: Is this the only option now? Change the const name and remove the `if`, if so.

**Review by rart — CHANGES_REQUESTED** [review]

**Review by rart — APPROVED** [review]

**Pull request merged**